### PR TITLE
Scenario builder warnings too long

### PR DESCRIPTION
**Pull request by guilpier-code**
Warnings from scenario builder can be very long :
this happens when a cluster (thermal or renewable) is disabled and when some TS numbers are assigned to this cluster in the scenario builder.
As the cluster is disabled, Antares warns the user of all years where a TS number is set.
When this year collection grows, the size of the warning line grows accordingly.

This work fixes the issues #398 and #613 (both pointing the same problem)

### CODE REVIEW

File: src/libs/antares/study/scenario-builder/rules.cpp [hunk 1]
```diff
@@ -364,21 +364,22 @@ void Rules::sendWarningsForDisabledClusters()
...
-        // to. Notice that these scenarii could be less then 10, but are at least 1.
-        string listScenarii = to_string(*itv + 1);
+        // Listing the 10 first years for which the current cluster was given a specific TS number 
+        // in the scenario builder.
+        // Note that this list of years size could be less then 10, but are at least 1.
+        string listYears = to_string(*itv + 1);
         itv++;
-        for (; itv != scenariiForCurrentCluster.end(); itv++)
-            listScenarii += ", " + to_string(*itv + 1);
+        for (; itv != scenariiForCurrentCluster.end() && *itv + 1 <= 10; itv++)
```
Comment (flomnes) [inline src/libs/antares/study/scenario-builder/rules.cpp]: No message is displayed if a study contains e.g 50 years, but the 10 first years are disabled. I recommend using a container, then aggregate it into a string.

Comment (guilpier-code) [reply src/libs/antares/study/scenario-builder/rules.cpp]: You're right, changes done

File: src/libs/antares/study/scenario-builder/rules.cpp [hunk 2]
```diff
@@ -364,21 +364,22 @@ void Rules::sendWarningsForDisabledClusters()
...
-        // to. Notice that these scenarii could be less then 10, but are at least 1.
-        string listScenarii = to_string(*itv + 1);
+        // Listing the 10 first years for which the current cluster was given a specific TS number 
+        // in the scenario builder.
+        // Note that this list of years size could be less then 10, but are at least 1.
+        string listYears = to_string(*itv);
         itv++;
-        for (; itv != scenariiForCurrentCluster.end(); itv++)
-            listScenarii += ", " + to_string(*itv + 1);
+        for (int year_count = 1; itv != scenariiForCurrentCluster.end() && year_count < 10; itv++, year_count++)
```
Comment (flomnes) [inline src/libs/antares/study/scenario-builder/rules.cpp:372]: Starting at 1, you'll only get 9 years. Start at 0 for 10 ?

**Comment by flomnes** [comment]
Resolve conflicts, set number of displayed years to 10 (currently 9) and merge.

File: src/libs/antares/study/scenario-builder/rules.cpp (hunk 2 above)
Comment (guilpier-code) [reply src/libs/antares/study/scenario-builder/rules.cpp:372]: Before the loop **for**, **listYears** is filled with 1 year.
In the loop, **listYears** is filled with 9 more years.

**Pull request merged**